CMakeLists.txt

```diff
@@ -183,6 +183,9 @@ add_definitions(-DAVX512_FP32_WEIGHT_ONLY_NF4=true)
 # add_definitions(-DSTEP_BY_STEP_ATTN=true)
 add_definitions(-DUSE_SHM=true)
 option(XFT_BUILD_TESTS "Build xfastertransformer unit tests" OFF)
+if(XFT_BUILD_TESTS)
+add_definitions(-DUNDEBUG=true)
```

Comment:
why do we need UNDEBUG?

Reply:
When we use CMake release, the -DNDEBUG flag is included by default. This option would make all assert statements ineffective as one kind of optimization. I undefine this flag during unit testing (ut) to ensure that assert performs the necessary checks.

Reply:
This is also why in some environments the assert may fail, while in others it does not. The behavior of assert can vary depending on the compilation settings and the specific conditions.